ddtrace/ci/internal/encoding.py

```diff
@@ -0,0 +1,63 @@
+import os
+from typing import Any
+from typing import List
+from typing import TYPE_CHECKING
+import uuid
+
+import ddtrace
+from ddtrace.internal import compat
+from ddtrace.internal._encoding import ListBufferedEncoder
+from ddtrace.internal._encoding import packb
+from ddtrace.internal.encoding import _EncoderBase
+
+
+if TYPE_CHECKING:
+    from ddtrace.span import Span
+
+
+class AgentlessEncoderV1(ListBufferedEncoder):
+    """Encode spans for use in Agentless CI Visibility intake endpoint."""
+
+    content_type = "application/msgpack"
+
+    def __init__(self, max_size, max_item_size):
+        # type: (int, int) -> None
+        """Detects and stores default metadata values."""
+        self.metadata = {
+            "*": {
+                "language": "python",
+                "library_version": ddtrace.__version__,
+                "language_version": compat.PYTHON_VERSION,
+                "runtime-id": uuid.uuid4().hex,
```

Comment:
This id should be a global runtime id for when the library is loaded. One option would be in `ddtrace.config` or something similar that we agree on as a shared global configuration.

Should this id be maintained for child processes? Yes.

Can this runtime-id be added to the trace encoder?